File: Lab3/3.py
from math import log
x = int(input("Please, enter the x: "))

ans = []
max_pow = int(log(x, 3) + 1) + 1

for k in range(max_pow):
    for l in range(max_pow-k):
        for m in range(max_pow-k-l):
            number = 3**k * 5**l * 7**m
            if (number <= x) and not (number in ans):
                ans.append([number, k, l, m])

print("The answer:\n" + "\n".join(f"{number}\t{k} {l} {m}" for number, k, l, m in sorted(ans, key=lambda x: x[0])))

File: Lab3/3.cpp
#include <cmath>
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    std::vector<int> answer;
    int x;
    std::cout << "Please, enter x: " << std::endl;
    std::cin >> x;
    int maxPow = (int)(log(x) / log(3)) + 1;
    for (int k; k < maxPow; k++) {
        for (int l = 0; l < maxPow - k; l++) {
            for (int m = 0; m < maxPow - k - l; m++) {
                int number = pow(3, k) * pow(5, l) * pow(7, m);
                if ((number <= x) && (std::find(answer.begin(), answer.end(), number) == answer.end())) 
                    answer.push_back(number);
            }
        }
    }
    std::sort(answer.begin(), answer.end());
    std::cout << "The answer:" << std::endl;
    for (int i = 0; i < answer.size(); i++) {
        std::cout << answer[i] << std::endl;
    }
    return 0;
}